dune_client/base_client.py

```diff
@@ -28,6 +31,15 @@ def __init__(
         self.logger = logging.getLogger(__name__)
         logging.basicConfig(format="%(asctime)s %(levelname)s %(name)s %(message)s")
 
+    @classmethod
+    def from_env(cls) -> BaseDuneClient:
+        """
+        Constructor allowing user to instantiate a client from environment variable
+        without having to import dotenv or os manually
+        """
+        load_dotenv()
```

Comment:
but load_dotenv() shouldn't be needed?
even with it, the user must be aware of the DUNE_API_KEY and needs to set that env flag (or write it in the .env file).
So it doesn't seem to be gaining anything, it just adds dependencies and yet another level of indirection.

lets not do it?

Reply:
I can try to remove the use of `load_dotenv`, but from memory, I am pretty sure its necessary. At least, from what I recall, this is what forces the .env file to be sourced at runtime. 

If we are ok with assuming that users have manually sourced their own env files then this may not be necessary. Let me give it a try.

Reply:
Everything check out. I have removed the use of this in the production code (only used in e2e tests now).